taskcluster/ci/release-notify-testrail/kind.yml

```diff
@@ -0,0 +1,72 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+---
+loader: taskgraph.loader.transform:loader
+
+transforms:
+    - taskgraph.transforms.notify:transforms
+    - taskgraph.transforms.job:transforms
+    - taskgraph.transforms.task:transforms
+
+task-defaults:
+    description: Sends Slack message to release testers that Testrail Milestone was created.
+    worker-type: b-android
+    worker:
+        docker-image: {in-tree: ui-tests}
+        max-run-time: 120
+        env:
+            TESTRAIL_PROJECT_ID: '59' # Fenix Browser
+            TESTRAIL_TEST_SUITE_ID: '49319' # Test Automation Release Milestone
+    run:
+        use-caches: false
+        using: run-commands
+        secrets:
+            - name: project/mobile/ci/testrail
+              key: testrailCredentials
+              path: .testrail_credentials.json
+              json: true
+    routes:
+        - notify.slack-channel.G016BC5FUHJ.on-failed
+    scopes:
+        - queue:route:notify.slack-channel.G016BC5FUHJ # notify mobile-alerts-sandbox on failure
+        - notify:slack-channel:G016BC5FUHJ
+        - queue:route:notify.slack-channel.C02KDDS9QM9 # notify mobile-testeng on success
+        - notify:slack-channel:C02KDDS9QM9
+
+tasks:
+    create-milestone-focus:
+        dependencies:
+            ui-test-apk: ui-test-apk-focus-arm-beta
+        description: Create Testrail Milestone for Focus
+        run-on-tasks-for: [github-push]
+        run-on-git-branches: [releases_v]
+        run:
+            pre-commands:
+                # get-secrets is called from '..' directory so we need to cd into any directory to make it work
+                - ["cd", "focus-android"]
+            commands:
+                - [python3, "../taskcluster/scripts/testrail_main.py"]
+
+        worker:
+            env:
+                SHIPPING_PRODUCT: focus
+                TESTRAIL_PRODUCT_TYPE: Focus
+    
+    create-milestone-fenix:
+        dependencies:
+            ui-test-apk: ui-test-apk-fenix-arm-beta
+        description: Create Testrail Milestone for Fenix
+        run-on-tasks-for: [github-push]
+        run-on-git-branches: [releases_v]
+        run:
+            pre-commands:
+                # get-secrets is called from '..' directory so we need to cd into any directory to make it work
+                - ["cd", "fenix"]
+            commands:
+                - [python3, "../taskcluster/scripts/testrail_main.py"]
+        worker:
+            env:
+                SHIPPING_PRODUCT: fenix
+                TESTRAIL_PRODUCT_TYPE: Firefox
```

Comment:
I noticed these tasks are pretty similar save for a few minor changes. You could probably simplify this into a single definition that then gets split out in a transform.. but honestly this is more than good enough and if you really want to go further with this, you can do so in a follow-up.

Reply:
that's a great idea, thank you. I need to get more familiar with everything in general and that sounds like a great exercise as well. i'll create a follow-up story to do that.